analyzer/runtime/extract.go

```diff
@@ -375,6 +376,10 @@ func ExtractRound(blockHeader nodeapi.RuntimeBlockHeader, txrs []nodeapi.Runtime
 						if to, err = registerRelatedEthAddress(blockData.AddressPreimages, blockTransactionData.RelatedAccountAddresses, *ok); err != nil {
 							return fmt.Errorf("created contract: %w", err)
 						}
+						blockTransactionData.EVMContract = &EVMContractData{
+							Address:    to,
```

Comment:
> Did anyone check how other explorers do this?

I checked briefly, it's usually one of (or a combination of both):
- if the (ethereum) node the explorer is fetching data from supports the "tracing API", the explorer will trace each transaction (via `debug_traceTransaction` call) and obtain various additional details about the transaction (among others - internal transactions (which are contract-to-contract calls) and obtain needed data via there). However, iirc the oasis-sdk/evm module does not support this tracing API. Although, I believe that the upstream [EVM library does support tracing](https://github.com/search?q=repo%3Arust-blockchain%2Fevm%20tracing&type=code), so theoretically this might be possible to implement in future.
- if tracing is not supported: call [`evm.Code`](https://github.com/oasisprotocol/oasis-sdk/blob/75e9e523c62b2857c3c1b4aee27c86ecdf2a671d/client-sdk/go/modules/evm/evm.go#L20)  for every encountered address to check if it is a regular account or a contract account.

I checked this briefly (and had some vague memory from researching this in the past) and I think that's mostly it, but I could have missed something.

Reply:
Oh interesting, I didn't know this could happen. We're currently planning on expanding the evm tokens analyzer (or creating an async standalone evm analyzer) to fetch the runtime bytecode of evm contracts async. I think for now it makes sense to enqueue other encountered addresses as well; and use a filter/heuristic to reduce the number of accounts we check if necessary.

> theoretically this might be possible to implement in the oasis-sdk/evm module in future.
this would be great when it's added :)

Reply:
> filter/heuristic to reduce the number of accounts we check if necessary

Yeah, I think it will be enough to check every encountered address only once and remember whether it's a regular account or a contract account, so it shouldn't be that bad.

Reply:
Tracing would be nice to have but since it's a somewhat bigger investment, I'd say let's launch without and see how users react / how much demand there is.

Regardless, even if we were to go ahead and implement it _now_, it would still take a while before it would be available on the nodes. So `evm.Code()` scanning of all accounts it is for us, for the time being.

@Andrew7234 can you open another ticket? I guess that will be yet another (simple) analyzer.

Reply:
Sure, filed https://app.clickup.com/t/866adz948